zebra/zebra_nhg.c

```diff
@@ -2812,6 +2825,9 @@ struct nhg_hash_entry *zebra_nhg_proto_add(uint32_t id, int type,
 		 */
```

Comment:
the comment just above here says that "the depends... may still be used" - the we 'release_all'. that's sort of confusing to read: is the comment going to be incorrect with the change?

Reply:
No the comment is still correct. We dont free or decrement the refcnt for the depends yet because they could and most likely are still be used. Added in that new function call just removes the childrens backrefs to the to old parent so they can be set with the new one.

I updated the comment to clear that up hopefully.